pkg/pool-aclamm/contracts/AclAmmPool.sol

```diff
@@ -170,22 +179,51 @@ contract AclAmmPool is BalancerPoolToken, PoolInfo, Version, IBasePool, BaseHook
             balancesScaled18,
             _virtualBalances,
             _c,
-            _sqrtQ0,
+            _calculateCurrentSqrtQ0(),
             _lastTimestamp,
-            _centernessMargin
+            _centernessMargin,
+            block.timestamp,
+            _sqrtQ0State
         );
     }
 
+    /// @inheritdoc IAclAmmPool
     function getLastTimestamp() external view returns (uint256) {
         return _lastTimestamp;
     }
 
-    function _setIncreaseDayRate(uint256 increaseDayRate) internal {
-        _c = AclAmmMath.parseIncreaseDayRate(increaseDayRate);
+    /// @inheritdoc IAclAmmPool
+    function getCurrentSqrtQ0() external view override returns (uint256) {
+        return _calculateCurrentSqrtQ0();
+    }
+
+    /// @inheritdoc IAclAmmPool
+    function setSqrtQ0(
+        uint256 newSqrtQ0,
+        uint256 startTime,
+        uint256 endTime
+    ) external onlySwapFeeManagerOrGovernance(address(this)) {
+        _sqrtQ0State.startSqrtQ0 = _calculateCurrentSqrtQ0();
+        _sqrtQ0State.endSqrtQ0 = newSqrtQ0;
+        _sqrtQ0State.startTime = startTime;
+        _sqrtQ0State.endTime = endTime;
```

Comment:
We should emit an event in this function, every time we change the sqrtQ0. The event should have start and end time and values.

Besides, I'd keep an internal _setSqrtQ0 function, and call it from the external setSqrtQ0, so we can use the internal function in the constructor as well.